### DISCUSSION TITLE
Login page not loading on Firefox?

### DISCUSSION THREAD

**Discussion by eoin-oshea** [community/community#23377] [New to GitHub]
<p>I’m learning web development on a course, so I’m very new to Github, but for some reason any time I try to log in to Github on Firefox, the login page doesn’t even load. It’s just blank. A white page. There’s no error messages, and having a look on the inspect panel just reveals the empty page has nothing but a header on it. The page loads fine on Edge, and also on Chrome, but Chrome is still really quite resource intense, so I’d rather not use it. I’m not using a proxy, and emptying cookies does nothing. There’s no scripting errors from what I can tell. JavaScript is active across all pages, and I don’t have any Add-ons installed that would interfere with the page loading. Is there anyone who could help on here? I am really confused…</p>

**Comment by galaxyeden** [comment]
<p>Hi <a class="mention" href="/eoin-oshea">@eoin-oshea</a> and welcome to GitHub! This is very surprising - a lot of us here at GitHub use Firefox every day, and it sounds like you’ve covered the basic checks - proxy, cookies, add-ons.</p>
<p>What version of Firefox are you running, and does GitHub load in a private window? Also, could you please check the browser console for errors (<code>Tools</code> → <code>Web Developer</code> → <code>Browser Console</code>) and let me know if you see any?</p>
<p>Thanks! Allie</p>

**Comment by eoin-oshea** [comment]
<p>Thanks for the quick reply! The loading doesn’t work on private windows either, just checked.<br>
There was a few messages on the console - three video loading errors (for media resource globe-900.hevc.mp4, globe-500.hevc.mp4, and actions-autocomplete.hevc.mp4, I can’t list the whole message because each one has a web address), and the scripting error below:<br>
:stop_sign: Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).</p>

**Comment by eoin-oshea** [comment]
<p>The video errors I mentioned are all in the structure of:<br>
Specified “type” attribute of “video/mp4; codecs=hevc,mp4a.40.2” is not supported. Load of media resource <a href="https://github.githubassets.com/images/modules/site/home/actions-autocomplete.hevc.mp4" rel="noopener nofollow ugc">https://github.githubassets.com/images/modules/site/home/actions-autocomplete.hevc.mp4</a> failed. Trying to load from next <source> element.<br>
(where the video file name is replaced with the relevant video on the list, but no other value on the message is changed).<br>
I just tried to run the login link with the console open, and got this error, maybe that might help:<br>
:arrow_forward: GET <a href="https://github.com/login" class="inline-onebox" rel="noopener nofollow ugc">Sign in to GitHub · GitHub</a> [HTTP/2 406 Not Acceptable 398ms]<br>
:stop_sign: Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).<br>
As you can see, the error message is the same as the one on the previous page.</source></p>

**Comment by galaxyeden** [comment]
<p>Hi <a class="mention" href="/eoin-oshea">@eoin-oshea</a> you’re very welcome! Could you please take screenshots of <code>Tools</code> → <code>Page Info</code> → <code>Security</code> as well as <code>About Firefox</code>?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/9/a9788b5aaae21b0cb8f67ca20af77fff605e8f47.png" data-download-href="/uploads/short-url/obcXmYn9M3Xd1eKE5dzQiqbpBiL.png?dl=1" title="Security"><img src="https://user-images.githubusercontent.com/25503691/181076642-96b89a80-32da-46fa-ad56-e93624a72d6c.png" alt="Security" data-base62-sha1="obcXmYn9M3Xd1eKE5dzQiqbpBiL" width="520" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/9/a9788b5aaae21b0cb8f67ca20af77fff605e8f47_2_520x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/9/a9788b5aaae21b0cb8f67ca20af77fff605e8f47_2_780x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/9/a9788b5aaae21b0cb8f67ca20af77fff605e8f47_2_1040x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/9/a9788b5aaae21b0cb8f67ca20af77fff605e8f47_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Security</span><span class="informations">1206×1158 116 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/7/275fa8df74529f40c3a0a85109ca704371d342c6.jpeg" data-download-href="/uploads/short-url/5CjzpMsabYz0MS3J2y8YjEscX78.jpeg?dl=1" title="About Firefox"><img src="https://user-images.githubusercontent.com/25503691/181076644-d166c55d-81d2-49ab-b926-2b0a8c0a7ac4.jpeg" alt="About Firefox" data-base62-sha1="5CjzpMsabYz0MS3J2y8YjEscX78" width="689" height="376" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/7/275fa8df74529f40c3a0a85109ca704371d342c6_2_689x376.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/2/7/275fa8df74529f40c3a0a85109ca704371d342c6_2_1033x564.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/7/275fa8df74529f40c3a0a85109ca704371d342c6.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/7/275fa8df74529f40c3a0a85109ca704371d342c6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">About Firefox</span><span class="informations">1344×734 135 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by eoin-oshea** [comment]
<p>Dunno if this will be of help, from what I can see, these are largely the same as what you’ve shown.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/5/257d0c10feb0afd99bc8613d7d403f123b0cfd15.png" data-download-href="/uploads/short-url/5lDzSlZVeKsW3yEeRnj3MjURj7v.png?dl=1" title="aboutfirefox" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/85488457/181076645-f4acc6e0-55d0-4f80-a03e-ba9f1433f6d2.png" alt="aboutfirefox" data-base62-sha1="5lDzSlZVeKsW3yEeRnj3MjURj7v" width="690" height="383" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/5/257d0c10feb0afd99bc8613d7d403f123b0cfd15_2_690x383.png, https://aws1.discourse-cdn.com/github/original/3X/2/5/257d0c10feb0afd99bc8613d7d403f123b0cfd15.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/5/257d0c10feb0afd99bc8613d7d403f123b0cfd15.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/5/257d0c10feb0afd99bc8613d7d403f123b0cfd15_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">aboutfirefox</span><span class="informations">859×478 716 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by eoin-oshea** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/a/6ac371fa2942407bba50874b1eb7abcf9eb3541f.png" data-download-href="/uploads/short-url/fetlMvbOjfTdnTdQnpc5Bo27Lfp.png?dl=1" title="tools-pageinfo-security" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/85488457/181076646-bd43fc07-42e3-4e9e-a9bf-8bbb04daef7d.png" alt="tools-pageinfo-security" data-base62-sha1="fetlMvbOjfTdnTdQnpc5Bo27Lfp" width="522" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6ac371fa2942407bba50874b1eb7abcf9eb3541f_2_522x500.png, https://aws1.discourse-cdn.com/github/original/3X/6/a/6ac371fa2942407bba50874b1eb7abcf9eb3541f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/6/a/6ac371fa2942407bba50874b1eb7abcf9eb3541f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6ac371fa2942407bba50874b1eb7abcf9eb3541f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">tools-pageinfo-security</span><span class="informations">780×747 748 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
Sorry, retricted by newbie limits :sweat_smile:

**Comment by galaxyeden** [comment]
<p>Absolutely - it looks the same - my expectation with the error was that you had may have been caused by security software intercepting the site via an auto-configured proxy. Alas, it appears not.</p>
<p>I’d like you to next try <a href="https://support.mozilla.org/en-US/kb/troubleshoot-extensions-themes-to-fix-problems">Firefox in safe mode</a> to be absolutely certain there are no extensions impacting this - that’s the only place I can think this blocked code may be coming from. Thanks!</p>

**Comment by eoin-oshea** [comment]
<p>Still doesn’t work. Here’s the screenshot, with console open - I assume that the extra errors are because of the safemode.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/9/19f3b67bbf22bfb591d38fed102ff4e0e870802e.jpeg" data-download-href="/uploads/short-url/3HA5wqMuPBdE5Wv95sYUFMxDOKG.jpeg?dl=1" title="firefox-safemode-github" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/85488457/181076650-9829610e-7ad8-412f-ac47-e5e8409b11f1.jpeg" alt="firefox-safemode-github" data-base62-sha1="3HA5wqMuPBdE5Wv95sYUFMxDOKG" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/9/19f3b67bbf22bfb591d38fed102ff4e0e870802e_2_690x388.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/1/9/19f3b67bbf22bfb591d38fed102ff4e0e870802e_2_1035x582.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/1/9/19f3b67bbf22bfb591d38fed102ff4e0e870802e_2_1380x776.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/9/19f3b67bbf22bfb591d38fed102ff4e0e870802e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">firefox-safemode-github</span><span class="informations">1920×1080 464 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by eoin-oshea** [comment]
<p>And this is the still empty login page, including a very empty console.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/7/a7f99dda7e6b6b03c0083744f320b4895b348900.png" data-download-href="/uploads/short-url/nXYxoKtJ9EgUBzSLMseW7506h7q.png?dl=1" title="firefox-safemode-github-login" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/85488457/181076651-0c4bf315-529e-4328-9895-f8fcb8390c9f.png" alt="firefox-safemode-github-login" data-base62-sha1="nXYxoKtJ9EgUBzSLMseW7506h7q" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7f99dda7e6b6b03c0083744f320b4895b348900_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7f99dda7e6b6b03c0083744f320b4895b348900_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7f99dda7e6b6b03c0083744f320b4895b348900_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7f99dda7e6b6b03c0083744f320b4895b348900_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">firefox-safemode-github-login</span><span class="informations">1920×1080 137 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by galaxyeden** [comment]
<p>Very interesting, indeed. I’m not seeing the <code>406</code> error anymore, though - which is what I was trying to resolve, is it still in the console?</p>
<p>I’ll take a look at this further tomorrow if no one else has ideas in the meantime, but it’s very interesting that it’s only Firefox that is affected. Perhaps you can also try installing the latest <a href="https://www.mozilla.org/en-GB/firefox/channel/desktop/">Firefox Nightly</a>, not because it’s going to fix the problem but because it will run as a completely isolated instance (since it’s an unstable test build) so should rule out any problems with Firefox and its configuration.</p>

**Comment by eoin-oshea** [comment]
<p>The 406 error is still there on the ‘login page’ (or lack thereof), it’s the only entry on the console (highlighted on the right-hand side in blue. I’ll give Nightly a go, and see how that plays out.</p>

**Comment by eoin-oshea** [comment]
<p>You were right to check if it works on Nightly, because it does. Even logs in fine.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/5/a55b80598732e998b01e3918ff2c0f26ff77e63c.png" data-download-href="/uploads/short-url/nAOP38vXTJaKaatXUQuT6xshtI8.png?dl=1" title="Firefox-Nightly" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/85488457/181076654-3a1a8b94-a529-4da1-9232-bb82ea5fdbc4.png" alt="Firefox-Nightly" data-base62-sha1="nAOP38vXTJaKaatXUQuT6xshtI8" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a55b80598732e998b01e3918ff2c0f26ff77e63c_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a55b80598732e998b01e3918ff2c0f26ff77e63c_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a55b80598732e998b01e3918ff2c0f26ff77e63c_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a55b80598732e998b01e3918ff2c0f26ff77e63c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Firefox-Nightly</span><span class="informations">1920×1080 119 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
Soooo… the question is… what on earth is stopping it in the regular build? :thinking:

**Comment by airtower-luna** [comment]
<p>This looks like a really interesting problem! The <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.6">406 status code</a> says that the server didn’t have content matching one of the <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-5.3">content negotioation headers</a> (“Accept” and friends) in the request, assuming it’s used correctly.</p>
<p>If you go to the “Network” tab in the developer tools you should be able to see the request and response headers after selecting the request for <code>https://github.com/login</code>. I’m curious what the content negotiation headers look like, and in particularly how they compare between the two Firefox versions you tried.</p>

**Comment by galaxyeden** [comment, marked as answer]
<p>Oops! I missed the right side for that <code>406</code>! As far as I’m aware, this basically has to be an extension (but we ruled that out with safe mode) or a configuration issue (likely a cached certificate or something else security-related). Are you willing to try completely <a href="https://support.mozilla.org/en-US/kb/refresh-firefox-reset-add-ons-and-settings">resetting Firefox</a>?</p>

**Comment by eoin-oshea** [comment]
<p>I really dunno what that did, but it worked! Thank you very much!</p>

**Comment by eoin-oshea** [comment]
<p>Oh, dang, I didn’t check, sorry!</p>

**Comment by galaxyeden** [comment]
<p>Hey <a class="mention" href="/eoin-oshea">@eoin-oshea</a> you’re very welcome and I’m so glad that worked for you! We had reached the point it <em>had</em> to be a setting somewhere, probably related to certificate trust or security options I’d guess, so that’s the “wipe everything” option. I really don’t like doing it, especially since it means I won’t know for future what actually caused it - but I didn’t want to take up all your time having you try one setting after another!</p>
